rotkehlchen/api/rest.py

```diff
@@ -1555,6 +1557,107 @@ def process_history(
+        """
+        error_or_empty, events = self.rotkehlchen.events_historian.get_history(
+            start_ts=from_timestamp,
+            end_ts=to_timestamp,
+            has_premium=self.rotkehlchen.premium is not None,
+        )
+        if error_or_empty != '':
+            return wrap_in_fail_result(error_or_empty, status_code=HTTPStatus.CONFLICT)
+
+        settings = self.rotkehlchen.get_settings()
+        ignored_ids = self.rotkehlchen.data.db.get_ignored_action_ids(None)
+        debug_info = {
+            'events': [entry.serialize() for entry in events],
+            'settings': settings.serialize(),
+            'ignored_events_ids': {k.serialize(): v for k, v in ignored_ids.items()},
+        }
+        return _wrap_in_ok_result(debug_info)
+
+    def get_history_debug(
+            self,
+            from_timestamp: Timestamp,
+            to_timestamp: Timestamp,
+            async_query: bool,
+    ) -> Response:
+        if async_query is True:
+            return self._query_async(
+                command=self._get_history_debug,
+                from_timestamp=from_timestamp,
+                to_timestamp=to_timestamp,
+            )
+
+        response = self._get_history_debug(
+            from_timestamp=from_timestamp,
+            to_timestamp=to_timestamp,
+        )
+        status_code = _get_status_code_from_async_response(response)
+        result_dict = _wrap_in_result(result=response['result'], message=response['message'])
+        return api_response(result_dict, status_code=status_code)
+
+    def _import_history_debug(self, filepath: Union[FileStorage, Path]) -> Dict[str, Any]:
+        """Imports the PnL debug data for processing and report generation"""
+        json_importer = HistoryJSONImporter(self.rotkehlchen.data.db)
+        if isinstance(filepath, Path):
+            success, data = json_importer.import_history_debug(filepath=filepath)
+        else:
+            tmpfilepath = self.import_tmp_files[filepath]
+            success, data = json_importer.import_history_debug(filepath=tmpfilepath)
+            tmpfilepath.unlink(missing_ok=True)
+            del self.import_tmp_files[filepath]
+
+        if success is False:
+            return wrap_in_fail_result(
+                message=data['msg'],
+                status_code=HTTPStatus.CONFLICT,
+            )
+        dbevents = DBHistoryEvents(self.rotkehlchen.data.db)
+        events = dbevents.get_history_events(
+            filter_query=HistoryEventFilterQuery.make(),
+            has_premium=True,
+        )
+        self.rotkehlchen.accountant.process_history(
+            start_ts=data['data']['start_ts'],
+            end_ts=data['data']['end_ts'],
+            events=list(events),
+        )
+        return OK_RESULT
+
+    def import_history_debug(
+            self,
+            async_query: bool,
+            filepath: Union[FileStorage, Path],
+    ) -> Response:
+        if getattr(sys, 'frozen', False) is True:
```

Comment:
that's not that I meant. Not just availability. The code should not exist. It should not be present.